374 search service

lykhachov opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Related Pull Requests

<!-- related pull request placeholder -->
<!--- ^ DO NOT REMOVE placeholder above ^ -->
https://github.com/duhon/magento-docker/pull/27


### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/catalog-storefront#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/catalog-storefront#374
2. Fixes magento/catalog-storefront#401


### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Code Review Checklist (*)


- [ ] Story AC is completed
- [ ] proposed changes correspond to [Magento Technical Vision](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html)
- [ ] new or changed code is covered with web-api/integration tests (if applicable)
- [ ] no backward incompatible changes

Commit:
added gitignore

Commit:
added gitignore

Commit:
init repo

mslabko commented on app/code/Magento/Catalog/Api/CategoryListInterface.php:
```diff
@@ -0,0 +1,22 @@
+<?php
```
can we add required dependency via composer and do not copy them and for modules where we need only few of files - just copy-paster them to new namespace?

Commit:
composer dependencies fix

Commit:
created custom attribute dependencies

Commit:
clean ConfigReader.php

owlsmage committed:
374: Remove index related logic from elastisearch

owlsmage committed:
374: Remove eav config from data provider

owlsmage committed:
374: Clean Elasticsearch

Commit:
remed store model

owlsmage committed:
374: Clean Elasticsearch

Commit:
Merge branch '374-search-service' into 374-search-service-catalog-remove

owlsmage committed:
374: Clean Elasticsearch

Commit:
clean ConfigReader.php

Commit:
removed Catalog and Eav module

Commit:
Merge remote-tracking branch 'origin/374-search-service-catalog-remove' into 374-search-service-catalog-remove

owlsmage committed:
374: Clean Elasticsearch

owlsmage committed:
374: Revert change in Elasticsearch6

owlsmage committed:
374: Mark files as copy in Elasticsearch6

owlsmage committed:
374: Mark files as copy in Elasticsearch6

Commit:
Moved Magento_Store to Magento_SearchStorefrontStore
Removed Magento_Directory
Cleaned Magento_SearchStorefrontAdvancedSearch

Commit:
Merge branch '374-search-service' of github.com:lykhachov/storefront-search-ce into 374-search-service

 Conflicts:
	app/code/Magento/SearchStorefrontElasticsearch/Model/Adapter/FieldMapper/Product/FieldProvider/DynamicField.php

Commit:
fix

Commit:
added "copied" comment to advancedSearch

Commit:
Moved Store Interface to Model from Api

Commit:
Removed Group.php and GroupRepository.php

Commit:
Cleaned up SearchStorefrontStore module

owlsmage committed:
374: Mark files as copy in Elasticsearch

owlsmage committed:
Merge remote-tracking branch 'origin/374-search-service' into 374-search-service

Commit:
changed module name in di

Commit:
moved di configs

Commit:
removed @api and @since

Commit:
cleaned up

Commit:
added comments to proto file

Commit:
cleaned up, removed CommandList.php

Commit:
cleaned up

Commit:
cleaned up

Commit:
cleaned up setup folder

Commit:
removed lib folder

owlsmage committed:
374: Remove setup folder and dependency on it

owlsmage committed:
374: Remove unused classes from elasticsearch

Commit:
removed Suggestions

Commit:
Merge remote-tracking branch 'origin/374-search-service' into 374-search-service

owlsmage committed:
374: Fix composer.json of modules

owlsmage committed:
374: Update composer.json and README.md of Magento_SearchStorefront

owlsmage committed:
374: remove unnecessary comments from di.xml

owlsmage committed:
374: update composer.json and README.md of Magento_SearchStorefrontConfig and Magento_SearchStorefrontSearch modules

owlsmage committed:
374: remove unused elasticsearch classes

owlsmage committed:
374: remove usage of category stub

Commit:
cleaned di.xml to run on minimum config available

Commit:
added ES7

Commit:
removed ES6

Commit:
cleaned up

Commit:
removed test file

Commit:
removed ES6

Commit:
put back some prefs

Commit:
cleaned up

Commit:
cleaned up

Commit:
changed init config

Commit:
cleaned ES di.xml

Commit:
removed config.xml stub and dependencies

Commit:
cleaned up

Commit:
cleaned up

Commit:
changed fallback for price buket in env.php

Commit:
added "copied" comment
moved some code back to ES module

Commit:
cleaned up

mslabko commented on .gitignore:
```diff
@@ -0,0 +1,70 @@
+!/pub/media/.htaccess
+/pub/media/attribute/*
+!/pub/media/attribute/.htaccess
+/pub/media/analytics/*
```
need to remove exata directoeis that not relevant to this repo

Commit:
cleaned up

Commit:
replaced EAV stubs with sql queries

Commit:
added config command to cli

Commit:
clean up

Commit:
added composer.json to modules

Commit:
removed unused interface

Commit:
changes to di.xml

Commit:
added descriptions to composer.json files

Commit:
changes according failed tests

Commit:
changes according failed tests

Commit:
renamed bootstrap, autoload
changed all dependencies in modules for these files
changed Magento_Grpc module to SearchStorefrontGrpc and all corresponding files

Commit:
renamed composer files

Commit:
renamed bin/magento to bin/storefront

Commit:
Merge branch '374-search-service' into search-service-as-standalone-app

Commit:
removed search_engine.xml

Commit:
Merge branch '374-search-service' into search-service-as-standalone-app

mslabko committed:
fix monolith installation

Commit:
cleared/added preferences
